File: docs/WidgetMill/flutter/sdui_flutter/lib/src/builders/safe_area.dart
import 'package:flutter/widgets.dart';
import '../model/sdui_node.dart';
import '../renderer.dart';

/// SafeArea — recua o filho dos intrusos do sistema (notch, status bar).
Widget buildSafeArea(BuildContext context, SduiNode node, SduiRenderer r) {
  final p = node.props;
  return SafeArea(
    top: p['top'] as bool? ?? true,
    bottom: p['bottom'] as bool? ?? true,
    left: p['left'] as bool? ?? true,
    right: p['right'] as bool? ?? true,
    child: r.maybeRender(context, node.child) ?? const SizedBox.shrink(),
  );
}

File: docs/WidgetMill/packages/spec/src/nodes/safeArea.ts
import { z } from "zod";
import { NodeRef } from "../node-ref";

/** SafeArea — recua o filho dos intrusos do sistema (notch, status bar). */
export const SafeAreaNode = z.object({
  type: z.literal("safeArea"),
  props: z
    .object({
      top: z.boolean().optional(),
      bottom: z.boolean().optional(),
      left: z.boolean().optional(),
      right: z.boolean().optional(),
    })
    .default({}),
  child: NodeRef.optional(),
});
